### ISSUE TITLE
i<leader>w doesn't work like it should

### ISSUE THREAD

**Issue by unphased** [bkad/CamelCaseMotion#37]
My motion mapping is set to `,`, because my leader is already bound to other things. This is fine, but take note of this in my examples.

When cursor is on `t`: 

    SearchPosition& pos

i.e.:

    SearchPosi[t]ion& pos

running `diw` (delete regular vim word) results in 

    & pos

running `di,w` results in 

    Searchpos

instead, it should result in

    Search& pos

It seems like `i,w` is applying the operation until the next beginning of word, which is wrong because the `i` changes those semantics. Note however, this behavior is correct: When cursor is on `S`, the result is, as expected:

    Position& pos

it correctly removes the inner camelcase word. 

So basically removing inner camelcase word when said word is the LAST word is broken.

Thanks. BTW, https://github.com/vim-scripts/camelcasemotion does not have this problem.

**Comment by bkad** [comment]
Thanks for the detailed example, I was able to reproduce this issue to the letter. I'll take a closer look when I get the chance.

**Comment by bkad** [comment]
I made a small patch to address this case, to solve the more general problem requires a rethinking of the usage of iskeyword which I'll tackle at a later date. Thanks again for your report!

**Comment by unphased** [comment]
Nice. Thanks! 👍 

I guess I'm just a bit confused about how this plugin is so fundamentally different from Ingo's original plugin which handles this case properly and hasnt had a commit since 2011.

**Comment by unphased** [comment]
No worries, thank you so much for explaining! Gosh i wish everyone were as nice and approachable as you are.
